### PR TITLE
check for nan in measurments

### PR DESCRIPTION
**Pull request by wallacbe**

### CODE REVIEW

File: openhtf/util/validators.py [hunk 1]
```diff
@@ -87,6 +87,9 @@ def __init__(self, minimum=None, maximum=None):
       self.maximum = maximum
 
     def __call__(self, value):
+      # Check for nan
+      if value != value:
```
Comment (fahhem) [inline openhtf/util/validators.py]: `math.isnan(value)`

**Comment by wallacbe** [comment]
Global import of math was causing "AttributeError: 'NoneType' object has no attribute 'isnan".
I can import it within the class.

**Comment by fahhem** [comment]
Yeah, validators is one of those modules madsci did the replace trick. You can import the module at the top and pass it into `__init__`. It's a lot more changes, unfortunately.

File: openhtf/util/validators.py [hunk 2]
```diff
@@ -87,6 +87,10 @@ def __init__(self, minimum=None, maximum=None):
       self.maximum = maximum
 
     def __call__(self, value):
+      # Check for nan
+      import math
```
Comment (grybmadsci) [inline openhtf/util/validators.py]: copy the usage of `numbers`, `import math` at the top, pass it to `Validators.__init__()`, stash it on self in that constructor, and reference it here as `self.math_module.isnan()`.  Sorry this is tricky, it's done to provide `__getattr__` behavior at the module level, and this is actually the officially suggested way to do this.

**Comment by wallacbe** [comment]
This is tricky.  I passed math to class Validators  but math is needed in the nested class InRange.  I'm reading that the methods of a nested class cannot directly access the instance attributes of the outer class.  I have to pass the math module into InRange, but the constructor doesn't appear to be the correct place to pass it.

File: openhtf/util/validators.py (hunk 2 above)
Comment (fahhem) [inline openhtf/util/validators.py]: He makes a good point, the only way to intercept this the way things are is to override `__getattribute__` in `Validators`. I think this is the best place to do the import, considering.

**Comment by wallacbe** [comment]
In the original commit I used the following method to test for nan without math module.  Is this an option given the current issue?

a = 1.0
b = float('inf')
c = 0
d = float('nan')

print a != a
print b != b
print c != c
print d != d

Output:
False
False
False
True

**Comment by wallacbe** [comment]
Made updates with a decorator.  Tested this and it worked PTAL thanks

**Comment by grybmadsci** [comment]
Hm, this is still kinda messy.  The preferred way of doing what you're trying to do is done with the `RegexMatcher`.  Creating a helper function like that starts getting tedious, though; I think you may be able to stash `sys.modules` in `self._modules` in `Validators.__init__()` and then refer to all modules as dict elements therein (`self._modules['re']` or `self._modules['math']`).

**Comment by wallacbe** [comment]
made changes, PTAL

File: openhtf/util/validators.py [hunk 3]
```diff
@@ -147,5 +140,7 @@ def __ne__(self, other):
   def MatchesRegex(self, regex):
     return self.RegexMatcher(regex, self.re_module.compile(regex))
```
Comment (grybmadsci) [inline openhtf/util/validators.py]: need to switch this to use `self.modules` instead of `self.re_module`

File: openhtf/util/validators.py [hunk 4]
```diff
@@ -3,63 +3,49 @@
 Additional validators may be registered by passing them to the Register()
 method.  They can then be accessed directly as attributes on the validators
 module, and will typically be a type, instances of which are callable:
-
```
Comment (grybmadsci) [inline openhtf/util/validators.py]: let's leave the spacing of this as-is.

**Comment by grybmadsci** [comment]
couple minor comments but otherwise LGTM

File: openhtf/util/validators.py (hunk 3 above)
Comment (wallacbe) [inline openhtf/util/validators.py]: Done, not sure what happened with the formatting.

Done

**Comment by jettisonjoe** [comment]
Uh oh, travis is barfing. Looks like it has to do with the pickling stuff @grybmadsci added.

**Comment by grybmadsci** [comment]
oh, yea validators have to be pickle'able =/ this is going to be a real nuisance.  in the interest of just getting this fix in, you can do the `import math` inline like you had it (feel free to either leave the `sys.modules` conversion in or revert it, the important thing is you can't stash a copy of it on the validator itself).

**Comment by wallacbe** [comment]
math is imported inline, PTAL

**Comment by jettisonjoe** [comment]
Might have fixed #189.

**Pull request merged**